src/resolve/resolved-varlink.c

```diff
@@ -37,6 +37,24 @@ static void lookup_parameters_destroy(LookupParameters *p) {
         free(p->name);
 }
 
+static int dns_query_new_for_varlink(
+                Manager *m,
+                DnsQuery **ret,
+                DnsQuestion *question_utf8,
+                DnsQuestion *question_idna,
+                DnsPacket *question_bypass,
+                int ifindex,
+                uint64_t flags,
+                sd_varlink *link) {
+
+        int r;
+
+        r = dns_query_new(m, ret, question_utf8, question_idna, question_bypass, ifindex, flags);
+        if (r == -ENOANO)
+                return sd_varlink_error(link, "io.systemd.Resolve.DNSError", NULL);
```

Comment:
Maybe, better to introduce a new error code.
`"io.systemd.Resolve.DNSError"` -> `"io.systemd.Resolve.QueryRefused"`
Please also update src/shared/varlink-io.systemd.Resolve.c .